packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -532,22 +533,35 @@ export class SinglePageResource {
   }
 }
 
-type InternalState<M extends BaseMetadata> = Readonly<{
-  optimisticUpdates: readonly OptimisticUpdate<M>[];
-  permissionsByRoom: Record<string, Set<Permission>>;
+type RoomId = string;
+type QueryKey = string;
 
-  // TODO: Ideally we would have a similar NotificationsDB, like we have ThreadDB
-  notificationsById: Record<string, InboxNotificationData>;
-  settingsByRoomId: Record<string, RoomNotificationSettings>;
-  versionsByRoomId: Record<string, Record<string, HistoryVersion>>;
-}>;
+/**
+ * Versions by roomId
+ * e.g. { 'room-abc': {versions: "all versions"}}
+ */
+type VersionsByRoomId = Record<RoomId, Record<string, HistoryVersion>>;
+
+type NotificationsById = Record<string, InboxNotificationData>;
 
 /**
- * Externally observable state of the store, which will have:
- * - Optimistic updates applied
- * - All deleted threads removed from the threads list
+ * Notification settings by room ID.
+ * e.g. { 'room-abc': { threads: "all" },
+ *        'room-def': { threads: "replies_and_mentions" },
+ *        'room-xyz': { threads: "none" },
+ *      }
  */
-export type UmbrellaStoreState<M extends BaseMetadata> = {
+type SettingsByRoomId = Record<RoomId, RoomNotificationSettings>;
+
+type PermissionHintsByRoomId = Record<RoomId, Set<Permission>>;
+
+export type CleanThredifications<M extends BaseMetadata> =
```

Comment:
```suggestion
export type CleanThreadifications<M extends BaseMetadata> =
```
no?

Reply:
Sure! Will update the name of this contraction!